src/main/java/uk/gov/companieshouse/web/accounts/service/company/impl/CompanyServiceImpl.java

```diff
@@ -128,16 +128,15 @@ private boolean isSameYearFiler(CompanyProfileApi companyProfile) {
     @Override
     public List<LocalDate> getFutureDatesForArd(LocalDate periodEndOn) {
 
+    	LocalDate todaysDate = LocalDate.now();
         List<LocalDate> futureValidDates = new ArrayList<>();
 
-
-        futureValidDates.add(periodEndOn.plusDays(1));
-        futureValidDates.add(periodEndOn.plusDays(2));
-        futureValidDates.add(periodEndOn.plusDays(3));
-        futureValidDates.add(periodEndOn.plusDays(4));
-        futureValidDates.add(periodEndOn.plusDays(5));
-        futureValidDates.add(periodEndOn.plusDays(6));
-        futureValidDates.add(periodEndOn.plusDays(7));
+        for(int i = 1; i <= 7; i++) {
+        	if(periodEndOn.plusDays(i).isBefore(todaysDate) ||
```

Comment:
How about flipping this logic:
```
if (periodEndOn.plusDays(i).isAfter(todaysDate)) {
  break; // break to bail out of the loop early to improve performance
}
futureValidDates.add(periodEndOn.plusDays(i));
```

Reply:
Sounds a reasonable switch around. Will update